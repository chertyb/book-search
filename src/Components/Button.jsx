import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const { text, handleClick } = props;
  return <button type="button" onClick={handleClick}>{text}</button>;
};

Button.propTypes = {
  text: PropTypes.string,
  handleClick: PropTypes.func,
};

Button.defaultProps = {
  text: '',
  handleClick: () => {},
};

export default Button;
