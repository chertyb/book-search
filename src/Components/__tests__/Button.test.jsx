import React from 'react';
import ReactDom from 'react-dom';
import { shallow } from '../../enzyme';
import Button from '../Button';

describe('Button component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<Button />, div);
    expect(div).toMatchSnapshot();
    ReactDom.unmountComponentAtNode(div);
  });

  it('should display props value', () => {
    const text = 'Search';
    const wrapper = shallow(<Button text={text} />);
    expect(wrapper.text()).toEqual(text);
  });
});
