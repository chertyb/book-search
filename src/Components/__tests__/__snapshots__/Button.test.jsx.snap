// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders without crashing 1`] = `
<div>
  <button
    type="button"
  />
</div>
`;
